src/librbd/io/ImageDispatcher.cc

```diff
@@ -193,11 +240,24 @@ bool ImageDispatcher<I>::send_dispatch(
     image_dispatch_spec->tid = ++m_next_tid;
   }
 
+  bool finished = preprocess(image_dispatch_spec);
```

Comment:
Nit: only when `tid == 0` right? No need to invoke preprocess for each continued layer of the image dispatch.

Reply:
I am sorry that I missed the point. I have updated it. Thank you.